### PR TITLE
fix: to make the app run on android device

### PR DESCRIPTION
**Pull request by ami-aman**
Ami app was not working on the android devices because of some missing react native files & functions. This PR has fixes & updates to make the app run on android.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
 
 buildscript {
     ext {
-        buildToolsVersion = "29.0.3"
+        buildToolsVersion = "31.0.0"
         minSdkVersion = 21
-        compileSdkVersion = 30
-        targetSdkVersion = 30
+        compileSdkVersion = 31
+        targetSdkVersion = 31
```
Comment (mrehan27) [inline android/build.gradle:10]: We just realized that our Android SDK doesn't work well on Android 12, so let's keep `targetSdkVersion` to `30` till we fix all issues for Android 12

File: android/build.gradle [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
+        classpath("com.facebook.react:react-native-gradle-plugin")
+        classpath("de.undercouch:gradle-download-task:5.0.1")
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
     }
 }
 
 allprojects {
     repositories {
-        mavenLocal()
```
Comment (mrehan27) [inline android/build.gradle:24]: We might need it for local testing of our SDK. So if possible, let' keep it there as follows.
```
ext.IS_DEVELOPMENT = System.getenv("CI") == null
if (IS_DEVELOPMENT) mavenLocal()
```

**Review by mrehan27 — CHANGES_REQUESTED** [review]

File: android/build.gradle (hunk 1 above)
Comment (ami-aman) [reply android/build.gradle]: fixed. Changed `targetSdkVersion` to 30

File: android/build.gradle (hunk 2 above)
Comment (ami-aman) [reply android/build.gradle:24]: fixed this. Thanks for mentioning.

**Review by mrehan27 — APPROVED** [review]

**Pull request merged**